src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -977,10 +961,16 @@ public void SetCookies(Uri uri, string cookieHeader)
         }
     }
 
-    internal sealed class PathList
+    internal struct PathList
     {
-        private readonly SortedList<string, CookieCollection> _list = new SortedList<string, CookieCollection>(PathListComparer.StaticInstance);
-        private readonly object _lock = new object();
+        private readonly SortedList<string, CookieCollection> _list;
+
+        public static PathList Create() => new PathList(new SortedList<string, CookieCollection>(PathListComparer.StaticInstance));
+
+        private PathList(SortedList<string, CookieCollection> list)
+        {
+            _list = list;
```

Comment:
I realize this is currently private and it's only called from the Create method, but consider adding a Debug.Assert(list != null) here as well.